### ISSUE TITLE
Speed up through qemu disk cache = unsafe !?

### ISSUE THREAD

**Issue by plajjan** [plajjan/vrnetlab#88]
So I look at one of my CI machines when I spin up a bunch of XRv routers and saw this:
```
avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.88    0.00    4.62   30.84    0.00   62.66

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
sda               0.00     0.00    0.67    0.00     4.00     0.00    12.00     0.00    0.00    0.00    0.00   0.00   0.00
sdb               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdc               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdd               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sde               0.00    21.00    0.00    0.67     0.00    86.67   260.00     0.00    0.00    0.00    0.00   0.00   0.00
sdf               0.00    61.67    0.00  881.33     0.00 482374.67  1094.65  1774.28 1849.44    0.00 1849.44   1.14 100.13
dm-0              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-1              0.00     0.00    0.00 1198.67     0.00 720526.67  1202.21  1692.08 1308.89    0.00 1308.89   0.84 100.13

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.76    0.00    3.68   34.82    0.00   59.73

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
sda               0.00     0.67    0.00    0.67     0.00     5.33    16.00     0.00    0.00    0.00    0.00   0.00   0.00
sdb               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdc               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdd               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sde               0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
sdf               0.00    28.67    0.00  800.00     0.00 482824.00  1207.06  1731.34 2169.74    0.00 2169.74   1.25  99.87
dm-0              0.00     0.00    0.00    0.00     0.00     0.00     0.00     0.00    0.00    0.00    0.00   0.00   0.00
dm-1              0.00     0.00    0.00  511.00     0.00 338582.67  1325.18  1676.53 3356.99    0.00 3356.99   1.95  99.87
```
There's quite a lot of disk IO going on there. It is rather bursty but I can't help but wonder a) what kind of disk access is this? Why is XRv writing so much data? and b) can we speed this up? 

I don't particularly care about safety of this data so perhaps using qemu/KVMs disk cache option and set it to `unsafe` would be ok!? We can see if the speed up is measurable, if not let's leave it but if there is a significant improvement we can consider turning of the cache.

**Comment by emjemj** [comment]
It should be fine, if the router dies it dies anyways. Sounds like a good idea.

**Comment by emjemj** [comment]
@plajjan did you try this out?

**Comment by plajjan** [comment]
@emjemj nope, I have not :(

**Comment by plajjan** [comment]
It seems the underlying problem is how our use of qcow images interact with the aufs image overlay backend in docker. I've analysed that issue and have a potential fix - see #144. With that merged the amount of data we write to disk will be insignificant so I don't think there is any actual benefit of doing cache=unsafe. It will just bring unsafety to this which is not necessary. Closing.
